### PR TITLE
[REF] web: convert core>utils>hooks tests to hoot 

### PR DESCRIPTION
**Pull request by SimonGenin**
Note for reviewer: most of useAutofocus are tagged as desktop, but not all, so the tag is set by test and not on the suite.

### CODE REVIEW

File: addons/web/static/lib/hoot-dom/helpers/dom.js [hunk 1]
```diff
@@ -330,6 +330,10 @@ const matchesQuery = (query, width, height) =>
  * @param {number} height
  */
 const matchesQueryPart = (query, width, height) => {
+    const pointerCoarse = query.match(/pointer:\s*coarse/);
+    if (pointerCoarse) {
+        return window.ontouchstart !== undefined;
+    }
```
Comment (aab-odoo) [inline addons/web/static/lib/hoot-dom/helpers/dom.js:336]: @Arcasias

File: addons/web/static/tests/core/utils/hooks.test.js [hunk 2]
```diff
@@ -0,0 +1,587 @@
...
+            static template = xml`
+                <span>
+                    <input type="text" t-ref="autofocus" />
+                </span>
+            `;
+            static props = ["*"];
+            setup() {
+                this.inputRef = useAutofocus();
+                onMounted(() => {
+                    // There is a ref, but the input is not focused
```
Comment (aab-odoo) [inline addons/web/static/tests/core/utils/hooks.test.js:103]: assert the input isn't focused

File: addons/web/static/tests/core/utils/hooks.test.js [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+                }
+                this._super();
+            },
+        });
+
+        const component = await mountWithCleanup(MyComponent);
+        expect(document.activeElement).toBe(component.inputRef.el);
+    });
+
+    test("does not focus when screen has touch", async () => {
```
Comment (aab-odoo) [inline addons/web/static/tests/core/utils/hooks.test.js:147]: ok so the missing assertion of "returns also a ref when screen has touch" is here. Those two tests should probably have been merged

**Review by aab-odoo — APPROVED** [review]

**Comment by SimonGenin** [comment]
@aab-odoo should be ok now

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**